lib/Service/ImageService.php

```diff
@@ -0,0 +1,596 @@
+	private function getTextFilePublic(?int $textFileId, string $shareToken): ?File {
+		// is the file shared with this token?
+		try {
+			$share = $this->shareManager->getShareByToken($shareToken);
+			if ($share->getShareType() === IShare::TYPE_LINK) {
+				// shared file or folder?
+				if ($share->getNodeType() === 'file') {
+					$textFile = $share->getNode();
+					if ($textFile instanceof File) {
+						return $textFile;
+					}
+				} elseif ($share->getNodeType() === 'folder' && $textFileId !== null) {
+					$folder = $share->getNode();
+					if ($folder instanceof Folder) {
+						$textFile = $folder->getById($textFileId);
+						if (count($textFile) > 0 && $textFile[0] instanceof File) {
+							return $textFile[0];
+						}
+					}
+				}
+			}
+		} catch (ShareNotFound $e) {
+			return null;
+		}
+		return null;
+	}
+
+	/**
+	 * Download a file and write it to a resource
+	 *
+	 * @param string $url
+	 * @param $resource
+	 * @param array $params
+	 * @param string $method
+	 * @return array|string[]
+	 */
+	private function simpleDownload(string $url, $resource, array $params = [], string $method = 'GET'): array {
+		$client = $this->clientService->newClient();
+		try {
+			$options = [
+				// does not work with sink if SSE is enabled
+				// 'sink' => $resource,
+				// rather use stream and write to the file ourselves
+				'stream' => true,
+				'timeout' => 0,
+				'headers' => [
+					'User-Agent' => 'Nextcloud Text',
+				],
+			];
+
+			if (count($params) > 0) {
+				if ($method === 'GET') {
+					$paramsContent = http_build_query($params);
+					$url .= '?' . $paramsContent;
+				} else {
+					$options['body'] = json_encode($params);
+				}
+			}
+
+			if ($method === 'GET') {
+				$response = $client->get($url, $options);
+			} elseif ($method === 'POST') {
+				$response = $client->post($url, $options);
+			} elseif ($method === 'PUT') {
+				$response = $client->put($url, $options);
+			} elseif ($method === 'DELETE') {
+				$response = $client->delete($url, $options);
+			} else {
+				return ['error' => 'Bad HTTP method'];
+			}
+			$respCode = $response->getStatusCode();
+
+			$body = $response->getBody();
+			while (!feof($body)) {
+				// write ~5 MB chunks
+				$chunk = fread($body, 5000000);
+				fwrite($resource, $chunk);
+			}
+
+			return ['Content-Type' => $response->getHeader('Content-Type')];
+		} catch (ServerException | ClientException $e) {
+			//$response = $e->getResponse();
+			//if ($response->getStatusCode() === 401) {
+			$this->logger->warning('Impossible to download image: '.$e->getMessage(), ['app' => Application::APP_NAME]);
+			return ['error' => 'Impossible to download image'];
+		} catch (ConnectException $e) {
+			$this->logger->error('Connection error: ' . $e->getMessage(), ['app' => Application::APP_NAME]);
+			return ['error' => 'Connection error'];
+		} catch (Throwable | Exception $e) {
+			$this->logger->error('Unknown error: ' . $e->getMessage(), ['app' => Application::APP_NAME]);
```

Comment:
```suggestion
			$this->logger->error('Unknown error when downloading the file', ['exception' => $e]);
```

No need for specifying the app if the LoggerInterface is injected through DI as it will be resolved to an instance that has the app already set. Also losing the exception directly will preserve the trace which might be helpful.

Reply:
Ok, fixed in many places.